models/user.go

```diff
@@ -38,11 +39,33 @@ type User struct {
 // Validate vaildate all of the user's input
 func (u *User) Validate() error {
 	if u.Username == "" {
-		return errors.New("User username is empty")
+		return errors.New("Username cannot be empty")
+	}
+
+	m, err := regexp.MatchString("^[a-zA-Z0-9@._-]*$", u.Username)
+	if err != nil {
+		return err
+	}
+
+	if !m {
+		return errors.New("Username can only contain the following characters: a-z 0-9 @._-")
 	}
 
 	if u.Password == "" {
-		return errors.New("User password is empty")
+		return errors.New("Password cannot be empty")
+	}
+
+	m, err = regexp.MatchString("^[a-zA-Z0-9@._-]*$", u.Password)
+	if err != nil {
+		return err
+	}
+
+	if !m {
+		return errors.New("Password can only contain the following characters: a-z 0-9 @._-")
+	}
+
+	if len(u.Password) < 8 {
```

Comment:
This wasn't limited due to the fact some people may have short names i.e. Ann, Dan, Jo